pyfzf/pyfzf.py

```diff
@@ -22,38 +22,48 @@
 #
 # Author: Nagarjuna Kumarappan <nagarjuna.412@gmail.com>
 
-from plumbum import local, FG
-import tempfile
+from shutil import which
+from subprocess import Popen, PIPE
 import os
+import tempfile
+
 
 # constants
 FZF_URL = "https://github.com/junegunn/fzf"
 
 
 class FzfPrompt:
-    def __init__(self):
-        self.sh = local['sh']
-        try:
-            self.fzf = local['fzf']
-        except:
-            raise SystemError("Cannot find 'fzf' installed on PATH. ( {0} )".format(FZF_URL))
+    def __init__(self, executable_path=None):
+        if executable_path:
+            self.executable_path = executable_path
+        elif not which("fzf") and not self.executable_path:
+            raise SystemError(f"Cannot find 'fzf' installed on PATH. ({FZF_URL})")
+        else:
+            self.executable_path = "fzf"
 
     def prompt(self, choices=None, fzf_options="", delimiter='\n'):
         # convert lists to strings [ 1, 2, 3 ] => "1\n2\n3"
         choices_str = delimiter.join(map(str, choices))
         selection = []
-        with tempfile.NamedTemporaryFile() as input_file:
-            with tempfile.NamedTemporaryFile() as output_file:
-                # Create an temp file with list entries as lines
-                input_file.write(choices_str.encode('utf-8'))
-                input_file.flush()
-
-                # Invoke fzf externally and write to output file
-                self.sh['-c', f"cat {input_file.name} | fzf {fzf_options} > {output_file.name}"] & FG
+        if os.name == "nt":
```

Comment:
This can be simplified. All platforms can be supported using the subprocess interface, so we don't need a branch just for Windows.

Reply:
@nk412 Yes, this solution is not elegant, but it is a necessary. You can check it out. The solution with the subprocess module will not work on Linux, and the solution using os.system and tempfile will not work on Windows. Permission Error occurs due to platform differences. I decided to investigate the problem deeper and I managed to simplify the code. 

The prompt method may look like this, I've done short tests on two platforms and it works. If you are satisfied with this, I can make a commit 

`
def prompt(self, choices=None, fzf_options="", delimiter='\n'):
        # convert lists to strings [ 1, 2, 3 ] => "1\n2\n3"
        choices_str = delimiter.join(map(str, choices))
        selection = []

        with tempfile.NamedTemporaryFile(delete=False) as input_file:
            with tempfile.NamedTemporaryFile(delete=False) as output_file:
                # Create an temp file with list entries as lines
                input_file.write(choices_str.encode('utf-8'))
                input_file.flush()

        # Invoke fzf externally and write to output file
        os.system(f"{self.executable_path} {fzf_options} < {input_file.name} > {output_file.name}")

        # get selected options
        with open(output_file.name) as f:
            for line in f:
                selection.append(line.strip('\n'))

        os.unlink(input_file.name)
        os.unlink(output_file.name)

        return selection
`